From 8bb501cea5db621dd82315e1226192d751c165a0 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 18 Jul 2019 12:59:14 -0700
Subject: [PATCH] TOOLS-2293 'ls-missing-release-builds' tool cannot find
 headnode-joyent, mockcloud, sdc-system-tests builds (fix the script to use
 the new buildisprivate label) Reviewed by: Chris Burroughs
 <chris.burroughs@joyent.com> Approved by: Chris Burroughs
 <chris.burroughs@joyent.com>

---
 tools/releng/check-repos-for-release   |  5 ++---
 tools/releng/ls-missing-release-builds | 15 ++++++---------
 2 files changed, 8 insertions(+), 12 deletions(-)

diff --git a/tools/releng/check-repos-for-release b/tools/releng/check-repos-for-release
index 3a4231d..8f71262 100755
--- a/tools/releng/check-repos-for-release
+++ b/tools/releng/check-repos-for-release
@@ -189,9 +189,8 @@ fi
 
 if [[ -z "$TARGETS" ]]; then
     # gets a list of the 'mg' label, which is the jenkins job name.
-    TARGETS=$(jr list -H -l mg='*',release=true -o labels | json -ag mg | sort -u | \
-        grep -v sdcsso | \
-        grep -v sdcboot)
+    TARGETS=$(jr list -H -l mg,release -o labels | json -ag mg | sort -u | \
+        grep -v sdcsso)
 fi
 
 for targ in $TARGETS; do
diff --git a/tools/releng/ls-missing-release-builds b/tools/releng/ls-missing-release-builds
index 51077d7..5b2f7f5 100755
--- a/tools/releng/ls-missing-release-builds
+++ b/tools/releng/ls-missing-release-builds
@@ -86,22 +86,19 @@ fi
 
 targets="$*"
 if [[ -z "$targets" ]]; then
-    targets=$(jr list -H -l mg='*',release=true -o labels | json -ag mg public | sort -u \
-        | grep -v '^sdcsso' \
-        | grep -v '^sdcboot' \
-        | grep -v '^headnode-debug' \
-        | grep -v '^headnode-joyent-debug')
+    targets=$(jr list -H -l mg,release -o labels | json -ag mg buildisprivate | sort -u \
+        | grep -v '^sdcsso')
 fi
 
 # Ensure mls is setup properly at all.
 mls /Joyent_Dev/public/builds >/dev/null \
     || fatal "cannot list /Joyent_Dev/public/builds"
 
-echo "$targets" | while read target public; do
-    if [[ "$public" == "true" ]]; then
-        latest_mpath=/Joyent_Dev/public/builds/$target/$release-latest
-    else
+echo "$targets" | while read target buildisprivate; do
+    if [[ "$buildisprivate" == "true" ]]; then
         latest_mpath=/Joyent_Dev/stor/builds/$target/$release-latest
+    else
+        latest_mpath=/Joyent_Dev/public/builds/$target/$release-latest
     fi
 
     case $target in
-- 
2.21.0

